include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -358,7 +359,11 @@ class argument_parser
      */
     void parse()
     {
+        if (parse_was_called)
+            throw parser_design_error("The function parse() must only be called once!");
```

Comment:
Exactly. The error should be thrown when the DEVeleoper is still writing the code and never occur for the user. I don't think I can check this at compile time so I had to throw an exception that is not directed at the user of a tool but at the developer.

Reply:
it is not in detail because it is not directed to SeqAn Core Developers but to Users that develop a tool. Another example for a parser design error is 
```
parser.add_flag(mybool1, 'f', "flag", "desc");
parser.add_flag(mybool1, 'f', "flag2", "desc");
```
which throws a parser_design_error saying the short identifier **must be unique**.
Those "design" errors are in general checked at compile time but unfortunately, in the case of the argument_parser, these are runtime decisions.